lib/utils.ex

```diff
@@ -68,4 +70,139 @@ defmodule LangChain.Utils do
     end)
     |> Enum.join("; ")
   end
+
+  @type callback_data ::
```

Comment:
Pretty much all of this logic was common with the OpenAI code, so I factored it out.